src/dbnode/storage/namespace_readers.go

```diff
@@ -149,23 +151,25 @@ func newNamespaceReaderManager(
 	opts Options,
 ) databaseNamespaceReaderManager {
 	return &namespaceReaderManager{
-		filesetExistsAtFn: fs.DataFileSetExistsAt,
-		newReaderFn:       fs.NewReader,
-		namespace:         namespace,
-		fsOpts:            opts.CommitLogOptions().FilesystemOptions(),
-		bytesPool:         opts.BytesPool(),
-		logger:            opts.InstrumentOptions().Logger(),
-		openReaders:       make(map[cachedOpenReaderKey]cachedReader),
-		metrics:           newNamespaceReaderManagerMetrics(namespaceScope),
+		filesetExistsFn: fs.DataFileSetExists,
+		newReaderFn:     fs.NewReader,
+		namespace:       namespace,
+		fsOpts:          opts.CommitLogOptions().FilesystemOptions(),
+		bytesPool:       opts.BytesPool(),
+		logger:          opts.InstrumentOptions().Logger(),
+		openReaders:     make(map[cachedOpenReaderKey]cachedReader),
+		metrics:         newNamespaceReaderManagerMetrics(namespaceScope),
 	}
 }
 
 func (m *namespaceReaderManager) filesetExistsAt(
 	shard uint32,
 	blockStart time.Time,
 ) (bool, error) {
-	return m.filesetExistsAtFn(m.fsOpts.FilePathPrefix(),
-		m.namespace.ID(), shard, blockStart)
+	// TODO(juchan): get the actual volume here.
```

Comment:
Just to make sure, this is being tracked in an issue yeah?

Reply:
Also, might it make more sense for the reader to just figure out the latest complete and open that instead of taking volume everywhere?

Reply:
https://github.com/m3db/m3/issues/1732